agentcfg/model_test.go

```diff
@@ -0,0 +1,76 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package agentcfg
+
+import (
+	"crypto/md5"
+	"fmt"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestNewDoc(t *testing.T) {
+	t.Run("InvalidInput", func(t *testing.T) {
+		d, err := NewDoc([]byte("some string"))
+		assert.Error(t, err)
+		assert.Empty(t, d)
+	})
+
+	t.Run("EmptyInput", func(t *testing.T) {
+		d, err := NewDoc([]byte{})
+		require.NoError(t, err)
+		expectedDoc := Doc{
+			Settings: map[string]string{},
+			ID:       fmt.Sprintf("%x", md5.Sum([]byte{}))}
+		assert.Equal(t, &expectedDoc, d)
```

Comment:
@axw as shown by this test I changed the logic to return an empty `{}` and a md5 sum over an empty `[]byte` for the ID.

Reply:
Thank you!